python/ray/worker.py

```diff
@@ -1295,6 +1298,15 @@ def connect(node,
             worker.run_function_on_all_workers(function)
     worker.cached_functions_to_run = None
 
+    # Setup tracing here
+    if _internal_kv_get("tracing_startup_hook"):
```

Comment:
Is `_internal_kv` namespaced by job? If not, doesn't this mean that different jobs are going to overwrite each others' hooks?

Reply:
In this PR we are setting it globally per cluster (set by first ray.init or ray start --head), discussing more of the future approach offline...